### PR TITLE
Object refactor

### LINKED ISSUE #3: Gem refactor: Organization, endpoint add, testing
### Feature Categories:

- [x] Other: Refactor/update

**Describe the feature/Bug your gonna work on**
- Refactor code out of som_timer.rb file into a service file. Refactor methods to adhere to SRP. 
- Add RSpec testing suite. Achieve 100% coverage. 
- Add endpoints for "rests"

**Additional context**
Add any other context or screenshots about the feature.
- gems are fun!

### PR DESCRIPTION
**Pull request by sienna-kopf**
# SOM-Timer Ruby Gem Pull Request Form

## Description
Please include a summary of the change.
- Refactored code out of som_timer.rb into lib/som_timer/service.rb file. Service file manages api calls much like in other applications I've built. 
- Added endpoints for "rests" based on updates made to the BE for mood tracking! This includes a "rests" method (get all "rests") and a "create_rest" method (post request to create a rest). Both of these have been added to the ReadMe Documentation. 
- Created Rspec testing suite and wrote tests for all methods. 100% coverage! 
- Pushed gem to [Fancy Ruby Gems Page](https://rubygems.org/gems/som-timer) as version 0.0.1

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Has This Been Tested?
- [ ] No
- [x] Yes

Details:
- To run test suite run:
```
bundle install rspec
```

# Checklist:
- [x] My code has no unused/commented out code
- [x] My code has no console logs/pry sessions
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,35 +1,39 @@
 # Som Timer Ruby Gem:
 
-## Overview: 
-Ruby Gem to access the endpoints of the [Som Timer](https://som-timer.herokuapp.com/) API. 
+## Overview:
+Ruby Gem to access the endpoints of the [Som Timer](https://som-timer.herokuapp.com/) API.
 
-## Gem Installation : 
-To install: 
+## Gem Installation :
```
Comment (rwilliams659) [inline README.md:6]: This is so cool!

File: lib/som_timer.rb [hunk 2]
```diff
@@ -3,34 +3,41 @@
 require "json"
 
 class SomTimer
-  def self.perform_request(path)
-    url = "https://som-timer-be.herokuapp.com/api/#{path}"
-    response = Faraday.get(url)
-    hash = JSON.parse(response.body, symbolize_names: true)
-    puts hash
+  def self.hello_world
```
Comment (rwilliams659) [inline lib/som_timer.rb:6]: Ah!!! I love this!

**Review by rwilliams659 — APPROVED** [review]
Awesome work as always, Sienna!

File: lib/som_timer/service.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+      req.headers['Content-Type'] = 'application/json'
+      req.body = { "mood_rating_1": "#{mood_rating_1}",
+                   "mood_rating_2": "#{mood_rating_2}",
+                   "content_selected": "#{content_selected}",
+                   "focus_interval": "#{focus_interval}",
+                   "rest_interval": "#{rest_interval}" }.to_json
+    end
+    JSON.parse(response.body, symbolize_names: true)
+  end
+end
```
Comment (Chulstro) [inline lib/som_timer/service.rb:62]: Oh now this is really clever

**Review by Chulstro — APPROVED** [review]
I love the encapsulation that you've done here with the service

**Review by sciencefixion — APPROVED** [review]
So nice to see some elegant Ruby again!

**Pull request merged**